cmd/skopeo/sync.go

```diff
@@ -758,7 +774,20 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 			} else {
 				logrus.WithFields(fromToFields).Infof("Copying image ref %d/%d", counter+1, len(srcRepo.ImageRefs))
 				if err = retry.IfNecessary(ctx, func() error {
-					_, err = copy.Image(ctx, policyContext, destRef, ref, &options)
+					manifestBytes, err := copy.Image(ctx, policyContext, destRef, ref, &options)
+					if opts.digestFile != "" {
+						manifestDigest, err := manifest.Digest(manifestBytes)
+						if err != nil {
+							return err
+						}
+						outputStr := fmt.Sprintf("%s %s", manifestDigest.String(), transports.ImageName(destRef))
+						if counter < len(srcRepo.ImageRefs)-1 {
```

Comment:
Why is this now conditional?

Traditionally in UNIX, “a line” is some characters _and a terminating newline_. Not having the terminating newline makes the file inconvenient to use, e.g. a trivial `cat` has the final line mixed with the next shell prompt (or, worse, hidden by the prompt).